File: UnixLab3/program/Go/main.go
package main

import (
    "fmt"
    "os"
    "strconv"
)

func Sum(first, second float64) float64 {
    return first + second
}

func main() {
    if (len(os.Args) < 3) {
        fmt.Println("Not enough args to complete the program!")
        return
    }

    first, error1  := strconv.ParseFloat(os.Args[1], 64)
    second, error2 := strconv.ParseFloat(os.Args[2], 64)

    if (error1 != nil || error2 != nil) {
        fmt.Println("Parse error...")
        os.Exit(1)
    }

    fmt.Printf("%v + %v = %v\n", first, second, Sum(first, second))
}

File: UnixLab3/program/C++/main.cpp
#include <iostream>

using std::cout;
using std::endl;

double Sum(const double first, const double second) {
    return first + second;
}

int main(int argc, const char** argv) {
    if (argc < 3) {
        cout << "Not enough args to complete this program" << endl;
        return EXIT_SUCCESS;
    }

    char* error;
    const auto first  = strtod(argv[1], &error);
    const auto second = strtod(argv[2], &error);


    if (*error != 0 && !isspace((unsigned char)*error)) {
        cout <<  "Parse error..." << endl;
        return EXIT_FAILURE;
    }

    cout << first << " + " << second << " = " << Sum(first, second) << endl;
    return EXIT_SUCCESS;
}
